### PR TITLE
Add a basic blog and RSS feed

### PR DESCRIPTION
**Pull request by eloquence**
Does not yet cover archival of old posts. To add a post, simply create it in `src/news` in the `<ISODATE>-title.md` format, and rebuild.

This introduces our first use of Eleventy's [global data files](https://www.11ty.dev/docs/data-global/), to populate some site metadata for the RSS feed.

I've tested the feed against W3C's feed validator, as well as a text-based RSS reader via `ngrok`:
Screenshot from 2024-08-13 17-06-53

### CODE REVIEW

File: src/news.njk [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+permalink: /news/
+---
+<div class="wrapper article">
+  <h1>News</h1>
+  {% for post in collections.posts | reverse %}
+    <article>
+      <h2 class="post-title"><a href="{{ post.url }}">{{ post.data.title }}</a></h2>
+      <div class="post-time"><time datetime="{{ post.date | dateIso }}">{{ post.date | dateReadable }}</time></div>
+      
+      {% if post.templateContent | striptags | length < 3000 %}
```
Comment (eloquence) [inline src/news.njk:13]: This probably needs some tweaking.

Comment (eloquence) [reply src/news.njk]: Tweaked it to cut off at 1500, up to 1500.

File: src/news.njk [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+layout: base
+title: News
+permalink: /news/
+---
+<div class="wrapper article">
+  <h1>News</h1>
```
[the remark this answers is not in the document]
Comment (eloquence) [reply src/news.njk]: I'm agnostic on RSS icons (we don't have them on our other sites), but I've added one. As we redesign the site, I suggest we standardize on an icon library like FontAwesome; for now I've adapted an icon from the Feather library (with inline attribution).

File: src/news/2024-08-14-welcome.md
[the remark this answers is not in the document]
Comment (eloquence) [reply src/news/2024-08-14-welcome.md:1]: Good catch; I've tweaked the sitewide style a bit to make the footer color the default.

**Comment by harrislapiroff** [comment]
The feed implementation here looks good to me and validates in the W3C's validator

**Comment by apyrgio** [comment]
Sigh, I just realized that with the addition of the "News" section, we now have a too large navbar for mobile devices:

Screenshot from 2024-08-20 09-15-56

We should either introduce a burger menu, or convert the "Code" / "Mastodon" links to :octocat: and :elephant: (_mastodon_) icons respectively.

File: src/news/2024-08-14-welcome.md
[the remark this answers is not in the document]
Comment (apyrgio) [reply src/news/2024-08-14-welcome.md:1]: Thanks, I saw the change. It seems that the footer still moves to the middle of the page, but now the rest of the space is filled with the default color. That's better, and we'll soon add more content to this page, so the footer will move more to the bottom.

**Comment by eloquence** [comment]
Good catch. I'm not sure I see obvious benefits in the absolute positioning we're doing in the CSS, which seems to get in the way of letting the browser just handle wrapping cleanly, in the absence of a hamburger (I blame the hamburglar).

Pushed a small commit that demonstrates what I mean but I defer to Harris on whether that's an acceptable fix :). At least in FF/Chrome it seemed to look fine in desktop & mobile.

Screen Shot 2024-08-20 at 17 45 27

**Review by harrislapiroff — APPROVED** [review]
I think that's an acceptable solution until the redesign comes along :)

**Pull request merged**